pkg/operator/sync.go

```diff
@@ -395,6 +396,11 @@ func (optr *Operator) syncMachineConfigController(config *renderConfig) error {
 	if err != nil {
 		return err
 	}
+	// Propagate our binary version into the controller config to help
+	// suppress rendered config generation until a corresponding
+	// new controller can roll out too.
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1879099
+	cc.Annotations[daemonconsts.GeneratedByVersionAnnotationKey] = version.Raw
```

Comment:
wait, why isn't this CC created in the cluster? should this be when we create the CC in the operator (sync.go.syncRenderConfig)? when this code runs we're sure that whatever we're upgrading from doesn't have this version and the check is going to fail

Reply:
> wait, why isn't this CC created in the cluster?

"In the cluster"?  What do you mean by that?

> when this code runs we're sure that whatever we're upgrading from doesn't have this version and the check is going to fail

We're reading a file that lives inside our own binary via bindata.
As far as we can tell it was just done this way originally because most of the rest of the code here is templating stuff too.

I mean conceptually all of this could be inline Go structs too, I think it would be clearer in *this* case but eh.

> what's happening right now is that this never gets synced in etcd cause it's after the actual sync to the api

Good catch!   I think in most cases it was working because we'd end up re-syncing again and at that point we'd get the version set, but we clearly want to ensure we sync to N and not N-1.

Fixed.

Reply:
> "In the cluster"? What do you mean by that?

I guess I just meant what I wrote in the last sentence lol sorry about the confusion